api/src/main/java/io/opencensus/metrics/MetricOptions.java

```diff
@@ -0,0 +1,157 @@
+   *
+   * <p>Default value is {@code "1"}.
+   *
+   * @return the unit of the Metric.
+   */
+  public abstract String getUnit();
+
+  /**
+   * Returns the list of label keys for the Metric.
+   *
+   * <p>Default value is {@link Collections#emptyList()}.
+   *
+   * @return the list of label keys for the Metric.
+   */
+  public abstract List<LabelKey> getLabelKeys();
+
+  /**
+   * Returns the map of constant labels (they will be added to all the TimeSeries) for the Metric.
+   *
+   * <p>Default value is {@link Collections#emptyMap()}.
+   *
+   * @return the list of label keys for the Metric.
+   */
+  // TODO: add support for this and make it public.
+  abstract Map<LabelKey, LabelValue> getConstantLabels();
+
+  /**
+   * Returns a new {@link Builder} with default options.
+   *
+   * @return a new {@code Builder} with default options.
+   * @since 0.5
+   */
+  public static Builder builder() {
+    return new AutoValue_MetricOptions.Builder()
+        .setDescription("")
+        .setUnit("1")
+        .setLabelKeys(Collections.<LabelKey>emptyList())
+        .setConstantLabels(Collections.<LabelKey, LabelValue>emptyMap());
+  }
+
+  @AutoValue.Builder
+  public abstract static class Builder {
+
+    /**
+     * Sets the description of the Metric.
+     *
+     * @param description the description of the Metric.
+     * @return this.
+     */
+    public abstract Builder setDescription(String description);
+
+    /**
+     * Sets the unit of the Metric.
+     *
+     * @param unit the unit of the Metric.
+     * @return this.
+     */
+    public abstract Builder setUnit(String unit);
+
+    /**
+     * Sets the list of label keys for the Metric.
+     *
+     * @param labelKeys the list of label keys for the Metric.
+     * @return this.
+     */
+    public abstract Builder setLabelKeys(List<LabelKey> labelKeys);
+
+    /**
+     * Sets the map of constant labels (they will be added to all the TimeSeries) for the Metric.
+     *
+     * @param constantLabels the map of constant labels
+     * @return this.
+     */
+    // TODO: add support for this and make it public.
+    abstract Builder setConstantLabels(Map<LabelKey, LabelValue> constantLabels);
+
+    abstract MetricOptions autoBuild();
+
+    /**
+     * Builds and returns a {@code MetricOptions} with the desired options.
+     *
+     * @return a {@code MetricOptions} with the desired options.
+     * @since 0.20
+     * @throws NullPointerException if {@code description}, OR {@code unit} is null, OR {@code
+     *     labelKeys} is null OR any element of {@code labelKeys} is null, OR OR {@code
+     *     constantLabels} is null OR any element of {@code constantLabels} is null.
+     * @throws IllegalArgumentException if any {@code LabelKey} from the {@code labelKeys} is in the
+     *     {@code constantLabels}.
+     */
+    public MetricOptions build() {
+      MetricOptions options = autoBuild();
+      Utils.checkNotNull(options.getDescription(), "description");
```

Comment:
nit: this is not needed because AutoValue will automatically throw a `NullPointerException` if `null` is set for a field without `@Nullable` annotation.

Reply:
done.